src/Api/Dfe.Complete.Api/Controllers/ProjectsController.cs

```diff
@@ -347,6 +350,97 @@ public async Task<IActionResult> ListAllProjectsByTrustRefAsync([FromQuery] List
             var project = await sender.Send(request, cancellationToken);
             return Ok(project.Value?.ProjectModels ?? []);
         }
+
+        /// <summary>
+        /// Returns a list of Notes for a Project
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        [Authorize(Policy = "CanRead")]
+        [HttpGet]
+        [Route("Notes")]
+        [SwaggerResponse(200, "Notes for project", typeof(List<NoteDto>))]
+        [SwaggerResponse(404, "Project not found")]
+        public async Task<IActionResult> GetNotesByProjectIdAsync([FromQuery] GetNotesByProjectIdQuery request, CancellationToken cancellationToken)
+        {
+            var result = await sender.Send(request, cancellationToken);
+
+            if (!result.IsSuccess || result.Value is null)
+                return NotFound();
+
+            return Ok(result.Value);
+        }
+
+        /// <summary>
+        /// Create a new Note for a Project
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        [Authorize(Policy = "CanReadWrite")]
+        [HttpPost]
+        [Route("Notes")]
+        [SwaggerResponse(201, "Note ID", typeof(NoteId))]
+        [SwaggerResponse(404, "Note not found")]
+        public async Task<IActionResult> CreateProjectNoteAsync([FromBody] CreateNoteCommand request, CancellationToken cancellationToken)
+        {
+            var result = await sender.Send(request, cancellationToken);
+
+            if (!result.IsSuccess || result.Value is null)
+                return StatusCode(500, result.Error);
+
+            return Created("", new NoteId(result.Value.Value));
+        }
+
+        /// <summary>
+        /// Update a Note for a Project
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        [Authorize(Policy = "CanReadWriteUpdate")]
+        [HttpPut]
+        [Route("Notes")]
+        [SwaggerResponse(200, "Note ID", typeof(NoteId))]
+        [SwaggerResponse(401, "Unauthorized access")]
+        [SwaggerResponse(404, "Note not found")]
+        public async Task<IActionResult> UpdateProjectNoteAsync([FromBody] UpdateNoteCommand request, CancellationToken cancellationToken)
+        {
+            var result = await sender.Send(request, cancellationToken);
+
+            if (!result.IsSuccess || result.Value is null)
+            {
+                if (result.ErrorType == ErrorType.Unauthorized)
+                    return Unauthorized(result.Error);
+                return StatusCode(500, result.Error);
```

Comment:
Nothing wrong this code but personally prefer conditional operator check for 2nd if condition instead of multiple if conditions (ignore it if you want to keep multiple if condition)
```suggestion
               if (!result.IsSuccess || result.Value is null)
    return result.ErrorType == ErrorType.Unauthorized
        ? Unauthorized(result.Error)
        : StatusCode(500, result.Error);

```